rust-examples/src/patternmatching.rs

```diff
@@ -0,0 +1,36 @@
+enum QuadraticSolution {
+    NoSolution,
+    OneSolition(f64),
+    TwoSolutions(f64, f64),
+}
+
+fn solve_quadratic_equation(a: f64, b: f64, c: f64) -> QuadraticSolution {
+    let discriminant = b * b - 4.0 * a * c;
+    if discriminant < 0.0 {
```

Comment:
I know this is exactly the point of your code snippet but it feels weird that this is not a match statement :D

Reply:
good point.